clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,8 +497,22 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
-
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
```

Comment:
For OffsetCommitRequest, I have updated the patch so that server will send `Errors.NOT_COORDINATOR` in the response if there is KAFKA_STORAGE_ERROR. This is similar to how broker currently converts NOT_LEADER_FOR_PARTITION to NOT_COORDINATOR when handling OffsetCommitRequest. No change on the client side is needed.

For OffsetForLeaderEpoch, the server will send KAFKA_STORAGE_ERROR and the replica fetcher thread will retry if there is error -- it doesn't care which error it is. Thus no change is needed.

For ListOffsets, the server will send KAFKA_STORAGE_ERROR and the client will convert this error to StaleMetadataException which extends InvalidMetadataException. Thus no change is needed.